datumaro/tests/test_transforms.py

```diff
@@ -373,26 +338,25 @@ def __iter__(self):
         self.assertEqual(3, len(actual.get_subset('test')))
 
     def test_random_split_gives_error_on_wrong_ratios(self):
-        class SrcExtractor(Extractor):
-            def __iter__(self):
-                return iter([DatasetItem(id=1)])
+        source_dataset = Dataset.from_iterable([DatasetItem(id=1)])
 
         with self.assertRaises(Exception):
-            transforms.RandomSplit(SrcExtractor(), splits=[
+            transforms.RandomSplit(source_dataset, splits=[
                 ('train', 0.5),
                 ('test', 0.7),
             ])
 
         with self.assertRaises(Exception):
-            transforms.RandomSplit(SrcExtractor(), splits=[])
+            transforms.RandomSplit(source_dataset, splits=[])
 
         with self.assertRaises(Exception):
-            transforms.RandomSplit(SrcExtractor(), splits=[
+            transforms.RandomSplit(source_dataset, splits=[
                 ('train', -0.5),
                 ('test', 1.5),
             ])
 
     def test_remap_labels(self):
+        #! why this test brokes when I rewrite it on Dataset ?
```

Comment:
Please, clean the comment. This probably can be fixed with non-inplace implementation of the transform. Another way to fix is probably to make a dataset from the transform result.